### PR TITLE
Handle zombie workflow in HSM framework

### PR DESCRIPTION
**Pull request by yycptt**
## What changed?

- Handle zombie workflow in HSM framework

## Why?

- Zombie workflow can not be updated.

## How did you test it?

- Unit test

### CODE REVIEW

File: service/history/consts/const.go [hunk 1]
```diff
@@ -68,6 +68,8 @@ var (
 	ErrActivityTaskNotCancelRequested = serviceerror.NewInvalidArgument("unable to mark activity as canceled without activity being request canceled first")
 	// ErrWorkflowCompleted is the error to indicate workflow execution already completed
 	ErrWorkflowCompleted = serviceerror.NewNotFound("workflow execution already completed")
+	// ErrWorkflowZombie is the error to indicate workflow execution is in zombie state and cannot be updated
+	ErrWorkflowZombie = serviceerror.NewNotFound("workflow execution is in zombie state cannot be updated")
```
Comment (bergundy) [inline service/history/consts/const.go:72]: You can reuse `ErrStaleReference` here and use `fmt.Errorf` to provide details.

File: service/history/statemachine_environment_test.go [hunk 2]
```diff
@@ -337,26 +343,52 @@ func TestAccess(t *testing.T) {
...
 			},
 		},
+		{
+			name:                "write zombie",
+			accessType:          hsm.AccessWrite,
+			workflowState:       enumsspb.WORKFLOW_EXECUTION_STATE_ZOMBIE,
+			expectedSetRequests: 0,
+			accessor: func(n *hsm.Node) error {
+				return fmt.Errorf("accessor should not be called")
+			},
```
Comment (bergundy) [inline service/history/statemachine_environment_test.go:353]: nit: you could also cast the root node to a mutable state and manipulate it there but this is also fine.

**Review by bergundy — APPROVED** [review]

File: service/history/consts/const.go (hunk 1 above)
Comment (yycptt) [reply service/history/consts/const.go]: make sense. Updated.

File: service/history/statemachine_environment_test.go (hunk 2 above)
Comment (yycptt) [reply service/history/statemachine_environment_test.go:353]: hmm the zombie check is at the beginning, and not invoke the accessor if failed, similar to invalid ref.

**Pull request merged**